### PR TITLE
Add Consul docs routes

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/0/1202632199481616/f) 🎟️

## 🗒️ What

- Adds `"consul"` slug to development and preview beta environments
- Adds `rootDocsPaths` and `navigationHeaderItems` properties to `src/data/consul.json`
- Adds `src/content/consul/docs-landing.json` with some stubbed content from https://www.consul.io/docs
- Updates some Learn tests that were referencing Consul as a beta product

## 🤷 Why

Getting Consul on the way to be a beta section! 🚀

### Docs pages

- [ ] [/consul](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul)
- [ ] [/consul/docs](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/docs)
- [ ] [/consul/api-docs](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/api-docs)
- [ ] [/consul/commands](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/commands)

### Additional pages that should "just work"

- [ ] [/consul/downloads](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/downloads)
- [ ] [/consul/tutorials](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/tutorials)
- [ ] [/consul/tutorials/cloud-get-started](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/tutorials/cloud-get-started)
- [ ] [/consul/tutorials/cloud-get-started/get-started](https://dev-portal-git-ambconsul-docs-routes-hashicorp.vercel.app/consul/tutorials/cloud-get-started/get-started)

### CODE REVIEW

**Comment by ashleemboyer** [comment]
I believe after #716 is merged, this PR shouldn't need any more changes. Fingers crossed! 😊

File: src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts
Comment (ashleemboyer) [inline src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]: @kendallstrautman these tests were failing because Consul was being hardcoded as a non-beta product. If I understand these two test cases correctly, their purpose is to test that non-beta product URLs aren't rewritten. I don't think my changes proposed in this file match that.

Do you have thoughts on what these should be updated to? Maybe just hardcode a different product? I did this in the other file that had failing tests (updated Boundary to be a non-beta product test case): `src/lib/remark-plugins/rewrite-tutorial-links/__tests__/rewrite-tutorial-links.test.ts`.

File: src/content/consul/docs-landing/docs.json
Comment (ashleemboyer) [inline src/content/consul/docs-landing/docs.json:3]: Self TODO 📌 I'm going to fill in this content tomorrow based on what I see [on the .io page](https://www.consul.io/docs).

File: src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts
Comment (kendallstrautman) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]: Ah yes! I would just put in a different non-beta product slug for now.

Comment (kendallstrautman) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]:
```suggestion
				expected: '/packer/tutorials/kubernetes',
```

Comment (kendallstrautman) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]: One thing we should do here thought is check in the `ACCEPTED_DOCS_PATHNAMES` array in the `rewrite-tutorial-links` index file. To make sure that if there are unique consul subpaths in .io they are included here.

Comment (kendallstrautman) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]: see my change here for reference https://github.com/hashicorp/dev-portal/pull/727

Comment (kendallstrautman) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:101]: With a quick look a consulproject.io I don't see any new paths. So perhaps we are good here.

File: src/content/consul/docs-landing/docs.json
Comment (ashleemboyer) [reply src/content/consul/docs-landing/docs.json]: ✅ Done in https://github.com/hashicorp/dev-portal/pull/721/commits/31a638cb1c50456a23a9dc112c9860829e506785

File: src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts
Comment (ashleemboyer) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts]: I went ahead and updated both files to reference packer as a beta rather than consul. 😊

Comment (ashleemboyer) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts]: Resolve since this comment is now outdated, but feel free to make another comment in the latest diff if needed!

File: src/views/product-root-docs-path-landing/components/supported-icons.tsx
Comment (kendallstrautman) [inline src/views/product-root-docs-path-landing/components/supported-icons.tsx:14]: Nice ty for the alphabetization 💯

File: src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts
Comment (kendallstrautman) [inline src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts:70]: 💭 We may want to actually update this one so its using a non beta product still if we want to keep the test 100% the same. 
```
/vagrant
https://learn.hashicorp.com/vagrant
```

**Review by kendallstrautman — APPROVED** [review]
This looks great! i'll defer to other folks who know the docs setup better than myself for a more thorough review, but everything is rendering as I would expect. 🙇 👏

File: src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts
Comment (ashleemboyer) [reply src/views/product-tutorials-view/helpers/__tests__/detect-and-reformat-learn-url.test.ts]: Great call! Updated in 51180eaae2ca39b38396e8451d4830f296fec62c

File: config/base.json
Comment (BRKalow) [inline config/base.json:21]: 🔢 5️⃣

**Review by BRKalow — APPROVED** [review]
image

**Pull request merged**
